File: python/OpenAPITokenClient.py
# OpenAPITokenClient.py
import json
from typing import Dict, Any, Optional, Union
from urllib.parse import urljoin

# 导入之前翻译的 HttpClient
from HttpClient import HttpClient, HttpClientOption


class OpenAPITokenClient:
    """OpenAPI Token 认证客户端"""

    def __init__(self, base_url: str, token: str):
        """
        初始化 OpenAPI Token 客户端

        Args:
            base_url: API 基础URL
            token: 认证令牌
        """
        self.token = token
        self.base_url = base_url.rstrip('/') + '/'  # 确保以斜杠结尾

        # 创建 HTTP 客户端配置
        headers = {"token": token}
        option = HttpClientOption(header=headers)
        self.client = HttpClient(option)

    def get(self, url: str, params: Optional[Dict[str, Any]] = None) -> Any:
        """
        发送 GET 请求

        Args:
            url: 相对URL路径
            params: 查询参数（可选）

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 发送 GET 请求
        response = self.client.get(full_url)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def post(self, url: str, data: Dict[str, Any]) -> Any:
        """
        发送 POST 请求

        Args:
            url: 相对URL路径
            data: 请求数据

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 发送 POST JSON 请求
        response = self.client.post_json(full_url, data)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def post_form(self, url: str, data: Dict[str, Any]) -> Any:
        """
        发送 POST 表单请求

        Args:
            url: 相对URL路径
            data: 表单数据

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 发送 POST 表单请求
        response = self.client.post_form(full_url, data)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def put(self, url: str, data: Dict[str, Any]) -> Any:
        """
        发送 PUT 请求

        Args:
            url: 相对URL路径
            data: 请求数据

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出

        Note:
            由于 HttpClient 没有直接实现 PUT 方法，
            这里使用 POST 并添加 X-HTTP-Method-Override 头
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 添加方法重写头
        headers = {"X-HTTP-Method-Override": "PUT"}

        # 发送请求
        response = self.client.post_json(full_url, data, headers=headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def delete(self, url: str) -> Any:
        """
        发送 DELETE 请求

        Args:
            url: 相对URL路径

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 发送 GET 请求（添加 DELETE 指示）
        headers = {"X-HTTP-Method-Override": "DELETE"}
        response = self.client.get(full_url, headers=headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def close(self) -> None:
        """关闭连接"""
        if self.client:
            self.client.close()

    def __enter__(self):
        """上下文管理器入口"""
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        """上下文管理器退出"""
        self.close()


class OpenAPITokenClientV2(OpenAPITokenClient):
    """
    OpenAPI Token 认证客户端（增强版）
    支持更多认证方式和配置选项
    """

    def __init__(self,
                 base_url: str,
                 token: str,
                 timeout: Optional[int] = 30,
                 verify_ssl: bool = True,
                 proxy: Optional[str] = None,
                 auth_header_name: str = "Authorization",
                 auth_header_format: str = "Bearer {}"):
        """
        初始化 OpenAPI Token 客户端（增强版）

        Args:
            base_url: API 基础URL
            token: 认证令牌
            timeout: 超时时间（秒）
            verify_ssl: 是否验证 SSL 证书
            proxy: 代理地址（格式: "host:port"）
            auth_header_name: 认证头名称
            auth_header_format: 认证头格式
        """
        self.token = token
        self.base_url = base_url.rstrip('/') + '/'
        self.auth_header_name = auth_header_name
        self.auth_header_format = auth_header_format

        # 格式化认证头
        auth_header_value = auth_header_format.format(token)

        # 创建 HTTP 客户端配置
        headers = {
            auth_header_name: auth_header_value,
            "User-Agent": "OpenAPI-Token-Client/1.0",
            "Accept": "application/json"
        }

        option = HttpClientOption(
            header=headers,
            socket_timeout=timeout * 1000 if timeout else None,
            connect_timeout=timeout * 1000 if timeout else None,
            ignore_ssl=not verify_ssl,
            proxy_address=proxy
        )

        self.client = HttpClient(option)

    def refresh_token(self, new_token: str) -> None:
        """
        刷新认证令牌

        Args:
            new_token: 新的认证令牌
        """
        self.token = new_token

        # 更新认证头
        auth_header_value = self.auth_header_format.format(new_token)
        self.client.session.headers[self.auth_header_name] = auth_header_value

    def get_with_auth_type(self,
                           url: str,
                           auth_type: str = "token") -> Any:
        """
        发送 GET 请求（支持多种认证类型）

        Args:
            url: 相对URL路径
            auth_type: 认证类型 ("token", "bearer", "basic")

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 根据认证类型设置不同的头
        if auth_type == "bearer":
            headers = {"Authorization": f"Bearer {self.token}"}
        elif auth_type == "basic":
            # Base64 编码 token
            import base64
            encoded_token = base64.b64encode(self.token.encode()).decode()
            headers = {"Authorization": f"Basic {encoded_token}"}
        else:  # 默认使用 token
            headers = {"token": self.token}

        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 发送 GET 请求
        response = self.client.get(full_url, headers=headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def request(self,
                method: str,
                url: str,
                data: Optional[Dict[str, Any]] = None,
                params: Optional[Dict[str, Any]] = None) -> Any:
        """
        发送通用请求

        Args:
            method: HTTP 方法（GET, POST, PUT, DELETE等）
            url: 相对URL路径
            data: 请求体数据
            params: 查询参数

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 根据方法发送请求
        if method.upper() == "GET":
            response = self.client.get(full_url)
        elif method.upper() == "POST":
            response = self.client.post_json(full_url, data or {})
        elif method.upper() == "PUT":
            # 添加方法重写头
            headers = {"X-HTTP-Method-Override": "PUT"}
            response = self.client.post_json(full_url, data or {}, headers=headers)
        elif method.upper() == "DELETE":
            # 添加方法重写头
            headers = {"X-HTTP-Method-Override": "DELETE"}
            response = self.client.get(full_url, headers=headers)
        else:
            raise ValueError(f"不支持的 HTTP 方法: {method}")

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text


class TokenManager:
    """Token 管理器，支持自动刷新"""

    def __init__(self,
                 base_url: str,
                 token: str,
                 refresh_url: Optional[str] = None,
                 refresh_interval: int = 3600):  # 默认1小时刷新一次
        """
        初始化 Token 管理器

        Args:
            base_url: API 基础URL
            token: 初始令牌
            refresh_url: 令牌刷新接口URL
            refresh_interval: 刷新间隔（秒）
        """
        self.base_url = base_url
        self.current_token = token
        self.refresh_url = refresh_url
        self.refresh_interval = refresh_interval
        self.last_refresh_time = 0

        # 创建客户端
        self.client = OpenAPITokenClientV2(
            base_url=base_url,
            token=token,
            auth_header_name="Authorization",
            auth_header_format="Bearer {}"
        )

    def should_refresh(self) -> bool:
        """检查是否应该刷新令牌"""
        current_time = time.time()
        return (self.refresh_url and
                current_time - self.last_refresh_time > self.refresh_interval)

    def refresh(self) -> bool:
        """
        刷新令牌

        Returns:
            是否刷新成功
        """
        if not self.refresh_url:
            return False

        try:
            # 调用刷新接口
            response = self.client.post(self.refresh_url, {
                "token": self.current_token
            })

            # 假设响应中包含新令牌
            if isinstance(response, dict) and "access_token" in response:
                new_token = response["access_token"]
                self.current_token = new_token
                self.client.refresh_token(new_token)
                self.last_refresh_time = time.time()
                return True

        except Exception as e:
            print(f"刷新令牌失败: {e}")

        return False

    def get(self, url: str, auto_refresh: bool = True) -> Any:
        """
        发送 GET 请求（支持自动刷新令牌）

        Args:
            url: 相对URL路径
            auto_refresh: 是否自动刷新令牌

        Returns:
            响应数据
        """
        if auto_refresh and self.should_refresh():
            self.refresh()

        return self.client.get(url)

    def post(self, url: str, data: Dict[str, Any], auto_refresh: bool = True) -> Any:
        """
        发送 POST 请求（支持自动刷新令牌）

        Args:
            url: 相对URL路径
            data: 请求数据
            auto_refresh: 是否自动刷新令牌

        Returns:
            响应数据
        """
        if auto_refresh and self.should_refresh():
            self.refresh()

        return self.client.post(url, data)

    def close(self) -> None:
        """关闭连接"""
        self.client.close()

File: go/OpenAPITokenClient.go
package main

import (
	"bytes"
	"crypto/tls"
	"encoding/base64"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"strings"
	"time"
)

// TokenHttpClientOption 定义HTTP客户端配置选项
type TokenHttpClientOption struct {
	Header         map[string]string
	SocketTimeout  int
	ConnectTimeout int
	IgnoreSSL      bool
	ProxyAddress   string
}

// TokenHttpClient HTTP客户端
type TokenHttpClient struct {
	client *http.Client
	header map[string]string
}

// NewTokenHttpClient 创建新的HTTP客户端
func NewTokenHttpClient(option TokenHttpClientOption) *TokenHttpClient {
	transport := &http.Transport{}

	// 配置代理
	if option.ProxyAddress != "" {
		proxyURL, err := url.Parse(option.ProxyAddress)
		if err == nil {
			transport.Proxy = http.ProxyURL(proxyURL)
		}
	}

	// 配置SSL验证
	if option.IgnoreSSL {
		if transport.TLSClientConfig == nil {
			transport.TLSClientConfig = &tls.Config{}
		}
		transport.TLSClientConfig.InsecureSkipVerify = true
	}

	// 配置超时
	client := &http.Client{
		Transport: transport,
	}

	if option.ConnectTimeout > 0 {
		client.Timeout = time.Duration(option.ConnectTimeout) * time.Millisecond
	}

	return &TokenHttpClient{
		client: client,
		header: option.Header,
	}
}

// Get 发送GET请求
func (c *TokenHttpClient) Get(url string, headers map[string]string) (*http.Response, error) {
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return nil, err
	}

	// 设置请求头
	c.setHeaders(req, headers)

	return c.client.Do(req)
}

// PostJSON 发送JSON POST请求
func (c *TokenHttpClient) PostJSON(url string, data interface{}, headers map[string]string) (*http.Response, error) {
	jsonData, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}

	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, err
	}

	// 设置默认Content-Type
	req.Header.Set("Content-Type", "application/json")

	// 设置请求头
	c.setHeaders(req, headers)

	return c.client.Do(req)
}

// PostForm 发送表单POST请求
func (c *TokenHttpClient) PostForm(urlPath string, data map[string]string, headers map[string]string) (*http.Response, error) {
	// 创建表单数据
	var formBuf bytes.Buffer
	for k, v := range data {
		if formBuf.Len() > 0 {
			formBuf.WriteString("&")
		}
		formBuf.WriteString(url.QueryEscape(k))
		formBuf.WriteString("=")
		formBuf.WriteString(url.QueryEscape(v))
	}

	req, err := http.NewRequest("POST", urlPath, &formBuf)
	if err != nil {
		return nil, err
	}

	// 设置默认Content-Type
	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")

	// 设置请求头
	c.setHeaders(req, headers)

	return c.client.Do(req)
}

// Close 关闭客户端（Go的http.Client不需要显式关闭，这里留空作为兼容）
func (c *TokenHttpClient) Close() {
	// 无需操作，Go的http.Client会自动管理连接
}

// 设置请求头
func (c *TokenHttpClient) setHeaders(req *http.Request, headers map[string]string) {
	// 设置客户端默认头
	for k, v := range c.header {
		req.Header.Set(k, v)
	}

	// 设置请求特定头（会覆盖默认头）
	for k, v := range headers {
		req.Header.Set(k, v)
	}
}

// OpenAPITokenClient OpenAPI Token认证客户端
type OpenAPITokenClient struct {
	token   string
	baseURL string
	client  *TokenHttpClient
}

// NewOpenAPITokenClient 创建新的OpenAPITokenClient
func NewOpenAPITokenClient(baseURL, token string) *OpenAPITokenClient {
	headers := map[string]string{
		"token": token,
	}

	option := TokenHttpClientOption{
		Header: headers,
	}

	client := NewTokenHttpClient(option)

	// 确保baseURL以斜杠结尾
	if !strings.HasSuffix(baseURL, "/") {
		baseURL += "/"
	}

	return &OpenAPITokenClient{
		token:   token,
		baseURL: baseURL,
		client:  client,
	}
}

// Get 发送GET请求
func (c *OpenAPITokenClient) Get(urlPath string, params map[string]string) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, params)
	if err != nil {
		return nil, err
	}

	response, err := c.client.Get(fullURL, nil)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// Post 发送POST请求
func (c *OpenAPITokenClient) Post(urlPath string, data map[string]interface{}) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, nil)
	if err != nil {
		return nil, err
	}

	response, err := c.client.PostJSON(fullURL, data, nil)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// PostForm 发送POST表单请求
func (c *OpenAPITokenClient) PostForm(urlPath string, data map[string]string) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, nil)
	if err != nil {
		return nil, err
	}

	response, err := c.client.PostForm(fullURL, data, nil)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// Put 发送PUT请求
func (c *OpenAPITokenClient) Put(urlPath string, data map[string]interface{}) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, nil)
	if err != nil {
		return nil, err
	}

	headers := map[string]string{
		"X-HTTP-Method-Override": "PUT",
	}

	response, err := c.client.PostJSON(fullURL, data, headers)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// Delete 发送DELETE请求
func (c *OpenAPITokenClient) Delete(urlPath string) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, nil)
	if err != nil {
		return nil, err
	}

	headers := map[string]string{
		"X-HTTP-Method-Override": "DELETE",
	}

	response, err := c.client.Get(fullURL, headers)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// Close 关闭客户端
func (c *OpenAPITokenClient) Close() {
	c.client.Close()
}

// 构建完整URL
func (c *OpenAPITokenClient) buildURL(urlPath string, params map[string]string) (string, error) {
	// 处理相对路径
	parsedBaseURL, err := url.Parse(c.baseURL)
	if err != nil {
		return "", err
	}

	parsedPath, err := url.Parse(urlPath)
	if err != nil {
		return "", err
	}

	fullURL := parsedBaseURL.ResolveReference(parsedPath).String()

	// 添加查询参数
	if params != nil && len(params) > 0 {
		parsedURL, err := url.Parse(fullURL)
		if err != nil {
			return "", err
		}

		q := parsedURL.Query()
		for k, v := range params {
			q.Add(k, v)
		}
		parsedURL.RawQuery = q.Encode()
		fullURL = parsedURL.String()
	}

	return fullURL, nil
}

// 处理响应
func (c *OpenAPITokenClient) handleResponse(response *http.Response) (interface{}, error) {
	if response.StatusCode >= 400 {
		body, _ := io.ReadAll(response.Body)
		return nil, fmt.Errorf("HTTP %d: %s", response.StatusCode, string(body))
	}

	body, err := io.ReadAll(response.Body)
	if err != nil {
		return nil, err
	}

	// 尝试解析JSON
	var jsonData interface{}
	if err := json.Unmarshal(body, &jsonData); err == nil {
		return jsonData, nil
	}

	// 不是JSON则返回原始文本
	return string(body), nil
}

// OpenAPITokenClientV2 OpenAPI Token认证客户端（增强版）
type OpenAPITokenClientV2 struct {
	OpenAPITokenClient
	authHeaderName   string
	authHeaderFormat string
}

// NewOpenAPITokenClientV2 创建新的OpenAPITokenClientV2
func NewOpenAPITokenClientV2(baseURL, token string, timeout *int, verifySSL bool, proxy string, authHeaderName, authHeaderFormat string) *OpenAPITokenClientV2 {
	// 格式化认证头
	authHeaderValue := fmt.Sprintf(authHeaderFormat, token)

	headers := map[string]string{
		authHeaderName: authHeaderValue,
		"User-Agent":   "OpenAPI-Token-Client/1.0",
		"Accept":       "application/json",
	}

	socketTimeout := 0
	connectTimeout := 0
	if timeout != nil {
		socketTimeout = *timeout * 1000
		connectTimeout = *timeout * 1000
	}

	option := TokenHttpClientOption{
		Header:         headers,
		SocketTimeout:  socketTimeout,
		ConnectTimeout: connectTimeout,
		IgnoreSSL:      !verifySSL,
		ProxyAddress:   proxy,
	}

	client := NewTokenHttpClient(option)

	// 确保baseURL以斜杠结尾
	if !strings.HasSuffix(baseURL, "/") {
		baseURL += "/"
	}

	return &OpenAPITokenClientV2{
		OpenAPITokenClient: OpenAPITokenClient{
			token:   token,
			baseURL: baseURL,
			client:  client,
		},
		authHeaderName:   authHeaderName,
		authHeaderFormat: authHeaderFormat,
	}
}

// RefreshToken 刷新认证令牌
func (c *OpenAPITokenClientV2) RefreshToken(newToken string) {
	c.token = newToken
	authHeaderValue := fmt.Sprintf(c.authHeaderFormat, newToken)
	c.client.header[c.authHeaderName] = authHeaderValue
}

// GetWithAuthType 发送GET请求（支持多种认证类型）
func (c *OpenAPITokenClientV2) GetWithAuthType(urlPath string, authType string) (interface{}, error) {
	var headers map[string]string

	switch authType {
	case "bearer":
		headers = map[string]string{
			"Authorization": fmt.Sprintf("Bearer %s", c.token),
		}
	case "basic":
		encodedToken := base64.StdEncoding.EncodeToString([]byte(c.token))
		headers = map[string]string{
			"Authorization": fmt.Sprintf("Basic %s", encodedToken),
		}
	default:
		headers = map[string]string{
			"token": c.token,
		}
	}

	fullURL, err := c.buildURL(urlPath, nil)
	if err != nil {
		return nil, err
	}

	response, err := c.client.Get(fullURL, headers)
	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// Request 发送通用请求
func (c *OpenAPITokenClientV2) Request(method, urlPath string, data map[string]interface{}, params map[string]string) (interface{}, error) {
	fullURL, err := c.buildURL(urlPath, params)
	if err != nil {
		return nil, err
	}

	var response *http.Response

	switch strings.ToUpper(method) {
	case "GET":
		response, err = c.client.Get(fullURL, nil)
	case "POST":
		response, err = c.client.PostJSON(fullURL, data, nil)
	case "PUT":
		headers := map[string]string{
			"X-HTTP-Method-Override": "PUT",
		}
		response, err = c.client.PostJSON(fullURL, data, headers)
	case "DELETE":
		headers := map[string]string{
			"X-HTTP-Method-Override": "DELETE",
		}
		response, err = c.client.Get(fullURL, headers)
	default:
		return nil, fmt.Errorf("不支持的HTTP方法: %s", method)
	}

	if err != nil {
		return nil, err
	}
	defer response.Body.Close()

	return c.handleResponse(response)
}

// TokenManager Token管理器，支持自动刷新
type TokenManager struct {
	baseURL         string
	currentToken    string
	refreshURL      string
	refreshInterval int
	lastRefreshTime int64
	client          *OpenAPITokenClientV2
}

// NewTokenManager 创建新的TokenManager
func NewTokenManager(baseURL, token, refreshURL string, refreshInterval int) *TokenManager {
	if refreshInterval <= 0 {
		refreshInterval = 3600 // 默认1小时
	}

	client := NewOpenAPITokenClientV2(
		baseURL,
		token,
		nil,
		true,
		"",
		"Authorization",
		"Bearer {}",
	)

	return &TokenManager{
		baseURL:         baseURL,
		currentToken:    token,
		refreshURL:      refreshURL,
		refreshInterval: refreshInterval,
		lastRefreshTime: 0,
		client:          client,
	}
}

// ShouldRefresh 检查是否应该刷新令牌
func (m *TokenManager) ShouldRefresh() bool {
	if m.refreshURL == "" {
		return false
	}

	currentTime := time.Now().Unix()
	return currentTime-m.lastRefreshTime > int64(m.refreshInterval)
}

// Refresh 刷新令牌
func (m *TokenManager) Refresh() bool {
	if m.refreshURL == "" {
		return false
	}

	data := map[string]interface{}{
		"token": m.currentToken,
	}

	response, err := m.client.Post(m.refreshURL, data)
	if err != nil {
		fmt.Printf("刷新令牌失败: %v\n", err)
		return false
	}

	// 尝试解析响应获取新令牌
	responseMap, ok := response.(map[string]interface{})
	if !ok {
		return false
	}

	newToken, ok := responseMap["access_token"].(string)
	if !ok {
		return false
	}

	m.currentToken = newToken
	m.client.RefreshToken(newToken)
	m.lastRefreshTime = time.Now().Unix()
	return true
}

// Get 发送GET请求（支持自动刷新令牌）
func (m *TokenManager) Get(urlPath string, autoRefresh bool) (interface{}, error) {
	if autoRefresh && m.ShouldRefresh() {
		m.Refresh()
	}

	return m.client.Get(urlPath, nil)
}

// Post 发送POST请求（支持自动刷新令牌）
func (m *TokenManager) Post(urlPath string, data map[string]interface{}, autoRefresh bool) (interface{}, error) {
	if autoRefresh && m.ShouldRefresh() {
		m.Refresh()
	}

	return m.client.Post(urlPath, data)
}

// Close 关闭连接
func (m *TokenManager) Close() {
	m.client.Close()
}
